test/test_va_api_config_attribs.cpp

```diff
@@ -39,40 +40,106 @@ class VAAPIConfigAttribs
+                EXPECT_EQ(sAttrib.value, aAttrib.value);
+            }
+        }
+    }
 };
 
-TEST_F(VAAPIConfigAttribs, GetConfigAttribs)
+TEST_P(VAAPIConfigAttribs, GetConfigAttribs)
 {
-    doGetMaxValues();
+    // The driver must support creating a config without any user specified
+    // attributes, in which case the driver will use it's own defaults.  The
+    // app should be able to query those default attributes for such created
+    // config via vaQueryConfigAttributes.  Those default attribute values
+    // should be consistent with supported attribute values returned by
+    // vaGetConfigAttributes.
+
+    const VAProfile& profile        = ::testing::get<0>(GetParam());
+    const VAEntrypoint& entrypoint  = ::testing::get<1>(GetParam());
 
+    doGetMaxValues();
     doQueryConfigProfiles();
-    const Profiles& profiles = getSupportedProfileList();
-    ASSERT_FALSE(profiles.empty());
 
-    for (const auto& profile : profiles) {
+    if (doFindProfileInList(profile)) {
         doQueryConfigEntrypoints(profile);
-        const Entrypoints& entrypoints = getSupportedEntrypointList();
-        ASSERT_FALSE(entrypoints.empty());
+        if (doFindEntrypointInList(entrypoint)) {
+            // create config without attributes (i.e. use driver defaults)
+            createConfig(profile, entrypoint);
 
-        for (const auto& entrypoint : entrypoints) {
-            doGetMaxNumConfigAttribs();
-            doCreateConfigNoAttrib(profile, entrypoint);
+            // query default attributes from the config we just created
+            ConfigAttributes actual;
+            queryConfigAttributes(profile, entrypoint, actual);
 
-            // Once ConfigID is obtained, then the attrib list is populated as
-            // well.  This will confirm that the values returned by calling
-            // vaGetConfigAttributes do match
+            // we're done with the config
+            destroyConfig();
 
-            // make a copy since we want to set .value = 0
-            ConfigAttributes attribs = getQueryConfigAttribList();
-            for (auto& attrib : attribs) {
-                attrib.value = 0;
-            }
+            // copy the actual attributes and reset their values so we
+            // can get the supported values for them.
+            ConfigAttributes supported = actual;
+            std::for_each(supported.begin(), supported.end(),
+                [](VAConfigAttrib& s) { s.value = 0; });
 
-            doGetConfigAttributes(profile, entrypoint, attribs);
-            doCheckAttribsMatch(attribs);
-            doDestroyConfig();
+            // get supported config attribute values
+            getConfigAttributes(profile, entrypoint, supported);
+
+            // verify actual config attribute values are supported values
+            validateConfigAttributes(actual, supported);
+        } else {
+            // entrypoint not supported
+            doLogSkipTest(profile, entrypoint);
```

Comment:
Feature request: doLogSkipTest should be able to differentiate between unsupported Profile and unsupported entrypoint. Better, should be made generic to accept any types or arguments for skipped tests. Not sure if I should be raising a feature request bug for this :)

Reply:
Out of scope for current PR.  Please file feature request.

Reply:
https://github.com/01org/libva-utils/issues/99 opened as feature request bug.